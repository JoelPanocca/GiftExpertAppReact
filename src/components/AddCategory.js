import React, { useState } from 'react';
import PropTypes from 'prop-types'

export const AddCategory = ( {setCategories} ) => {
    const [newCategory, setNewCategory] = useState('')
    const handleOnSubmit = (e) => {
        e.preventDefault();
        if (newCategory.trim().length > 2) {
            setCategories( cats => [newCategory, ...cats]);
            setNewCategory('');
        }
        
    }

    const handleOnChange = (e) => {
        setNewCategory(e.target.value);
    }
    return (
        <form onSubmit={handleOnSubmit}>
            <input type = "text" value={newCategory} onChange={handleOnChange} placeholder="Escriba mÃ¡s de 3 letras y enter"/> 
        </form>
    )
}

AddCategory.propTypes = {
    setCategories: PropTypes.func.isRequired
}
