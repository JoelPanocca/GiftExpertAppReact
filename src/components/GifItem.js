import React from "react";
import PropTypes from 'prop-types';

export const GifItem = (img) => {
  return (
    <div className="card">
      <p>{img.titulo}</p>
      <img src={img.url} alt={img.titulo} />
    </div>
  );
};

GifItem.propTypes = {
    titulo : PropTypes.string.isRequired,
    url: PropTypes.string.isRequired
}
