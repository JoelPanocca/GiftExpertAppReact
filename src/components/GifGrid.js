import React from "react";
import { useFetchGif } from "../hooks/useFetchGif";
import { GifItem } from "./GifItem";

export const GifGrid = ({ category }) => {
  const {data:images, loading } = useFetchGif(category);
  return (
    <>
      <h2>{category}</h2>
      {loading && <p>Cargando...</p>}

      <div className="card-grid">
        {images.map((image) => {
          return <GifItem key={image.id} {...image} />;
        })}
      </div>
    </>
  );
};
