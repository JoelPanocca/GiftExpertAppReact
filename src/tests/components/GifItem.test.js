import {shallow} from 'enzyme'
import '@testing-library/jest-dom'
import { GifItem } from '../../components/GifItem';

describe('Test para GifGrid ITEMS', () => {
    const img = {titulo: 'Goku', url: 'https://goku.com/img.jpg'};
    const wrapper = shallow(<GifItem {...img} />);

    test('Debe evaluar correctamente contra su snapshot', () => {        
        expect(wrapper).toMatchSnapshot();
    })

    test('Debe evaluar la imagen igual a los props enviados', () => {
        const imgn = wrapper.find('img');
        expect ( imgn.prop('src')) .toBe (img.url);
        expect ( imgn.prop('alt')) .toBe (img.titulo);
    })
    
    
})
