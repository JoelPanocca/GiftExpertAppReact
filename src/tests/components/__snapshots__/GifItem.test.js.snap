// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test para GifGrid ITEMS Debe evaluar correctamente contra su snapshot 1`] = `
<div
  className="card"
>
  <p>
    Goku
  </p>
  <img
    alt="Goku"
    src="https://goku.com/img.jpg"
  />
</div>
`;
